test/miscellaneous_tests/nonlinear_solve.jl

```diff
@@ -20,67 +23,68 @@ let
-    @test isapprox(sol2[3], (p[5] * p[2]) / (p[6] * p[1]); atol=1e-10)
+    @test sol1[:X1] ≈ nlprob.ps[:k1] / nlprob.ps[:k2] atol=1e-10
+    @test sol1[:X2]^3 / factorial(3) ≈ nlprob.ps[:k3] / nlprob.ps[:k4] atol=1e-10
+    @test sol1[:X3] ≈ (nlprob.ps[:k5] * nlprob.ps[:k2]) / (nlprob.ps[:k6] * nlprob.ps[:k1]) atol=1e-10
+    @test sol2[:X1] ≈ nlprob.ps[:k1] / nlprob.ps[:k2] atol=1e-10
+    @test sol2[:X2]^3 / factorial(3) ≈ nlprob.ps[:k3] / nlprob.ps[:k4] atol=1e-10
+    @test sol2[:X3] ≈ (nlprob.ps[:k5] * nlprob.ps[:k2]) / (nlprob.ps[:k6] * nlprob.ps[:k1]) atol=1e-10
 end
 
 # Creates a system with multiple steady states.
 # Checks that corresponding ODEFunction return 0.0 in both cases. Checks for manually computed function as well.
-# Checks with SYmbol `u0` and Symbolic `p`.
 let 
-    # Creates steady state network, unpack the parameter values.
+    # Creates steady state network.
     steady_state_network_2 = @reaction_network begin
         v / 10 + hill(X, v, K, n), ∅ → X
         d, X → ∅
     end
-    @unpack v, K, n, d = steady_state_network_2
 
     # Creates NonlinearProblem.
-    u0 = [:X => 1.0]
-    p = [v => 1.0 + rand(rng), K => 0.8, n => 3, d => 1.0]
-    nl_prob = NonlinearProblem(steady_state_network_2, u0, p)
+    u0 = rnd_u0(steady_state_network_2, rng; min = 1.0)
+    ps = rnd_ps(steady_state_network_2, rng)
+    nlprob = NonlinearProblem(steady_state_network_2, u0, ps)
     
     # Solves it using standard algorithm and simulation based algorithm.
-    sol1 = solve(nl_prob; abstol=1e-12, reltol=1e-12).u
-    sol2 = solve(nl_prob, DynamicSS(Rosenbrock23()); abstol=1e-12, reltol=1e-12).u
+    sol1 = solve(nlprob; abstol=1e-12, reltol=1e-12)
+    sol2 = solve(nlprob, DynamicSS(Rosenbrock23()); abstol=1e-12, reltol=1e-12)
     
     # Computes NonlinearFunction (manually and automatically).
-    nfunc = NonlinearFunction(convert(NonlinearSystem, steady_state_network_2))    
+    nlprob_eval_1 = remake(nlprob; u0 = [:X => sol1[1]])
+    nlprob_eval_2 = remake(nlprob; u0 = [:X => sol2[1]])
     function nf_manual(u,p)
-        X = u[1]
+        X = u[:X]
         v, K, n, d = p
         return v/10 + v * X^n/(X^n + K^n) - d*X
     end
 
     # Tests solutions are correct.
-    @test isapprox(nfunc(sol1, last.(p))[1], 0.0; atol=1e-10)
-    @test isapprox(nfunc(sol2, last.(p))[1], 0.0; atol=1e-10)
-    @test isapprox(nf_manual(sol1, last.(p)), 0.0; atol=1e-10)
-    @test isapprox(nf_manual(sol2, last.(p)), 0.0; atol=1e-10)
+    @test nlprob_eval_1.f(nlprob_eval_1.u0, nlprob_eval_1.p)[1] ≈ 0.0 atol=1e-10
+    @test nlprob_eval_2.f(nlprob_eval_2.u0, nlprob_eval_2.p)[1] ≈ 0.0 atol=1e-10
+    @test nf_manual(sol1, last.(ps)) ≈ 0.0 atol=1e-10
+    @test nf_manual(sol2, last.(ps)) ≈ 0.0 atol=1e-10
 end
 
 # Checks for system with conservation laws.
 # Checks using interfacing with output solution.
 let 
+    # Conservation laws currently broken (you get stuck in an infinite loop in MTK or something).
+    return (@test_broken false)
```

Comment:
Another conservation law issue, this one only appears when a `XProblem` is created (because these creates a `MTKParameters` object, which is the real culprit, not sure what the problem in these are though).